pkg/skaffold/kubernetes/loader/load.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loader
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/reference"
+	"k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/config"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubectl"
+	kubectx "github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+type ImageLoader struct {
+	kubeContext string
+	cli         *kubectl.CLI
+}
+
+type Config interface {
+	GetKubeContext() string
+	LoadImages() bool
+}
+
+func NewImageLoader(kubeContext string, cli *kubectl.CLI) *ImageLoader {
+	return &ImageLoader{
+		kubeContext: kubeContext,
+		cli:         cli,
+	}
+}
+
+// We only load images that
+//    1) Were identified as local images by the Runner, and
+//    2) Are part of the set of images being deployed by a given Deployer, so we don't duplicate effort
```

Comment:
even if multiple deployers were using the same image and loaded them twice, we don't expect to get any errors, right?

Reply:
right, this will really only save us a veryyy small amount of computation, as subsequent calls to on the same image in skaffold will actually find the image in the target cluster node, and skip any calls to `load`.